===================================================================
RCS file: /repository/curl/lib/transfer.c,v
retrieving revision 1.182
retrieving revision 1.183
diff -p --unified=3 -r1.182 -r1.183
--- transfer.c	2003/10/24 21:54:34	1.182
+++ transfer.c	2003/11/06 07:55:45	1.183
@@ -1854,10 +1854,9 @@ CURLcode Curl_perform(struct SessionHand
          to the new URL */
       urlchanged = data->change.url_changed;
       if ((CURLE_OK == res) && urlchanged) {
-        char *gotourl;
         res = Curl_done(conn);
         if(CURLE_OK == res) {
-          newurl = strdup(data->change.url);
+          char *gotourl = strdup(data->change.url);
           res = Curl_follow(data, gotourl);
           if(res)
             free(gotourl);
